### ISSUE TITLE
add support for logarithmic scales

### ISSUE THREAD

**Issue by UnderworldSunny** [elide-b/starbars#12]
The current implementation does not support good log scaling: 
```
import starbars
import matplotlib.pyplot as plt

fig, ax = plt.subplots(figsize=(8,6))
# Example data
categories = ['A', 'B', 'C']
values = [6e3, 7e3, 9e3]
annotations = [('A', 'C', 0.0001),('B', 'C', 0.01),('A', 'C', 0.0001),('A', 'C', 0.0001),('B', 'C', 0.01)]
ax.set_yscale('log')
ax.bar(categories, values)
ax.set_ylim(1e2, None)

tiplength=0.05
textdistance=0.05
barmargin=0.05
topmargin=0.1

starbars.draw_annotation(annotations,fontsize=8, bar_margin=barmargin,top_margin=topmargin,text_distance=textdistance,tip_length=tiplength, ns_show=False)

plt.show()
```

this seems to privide good results 
```
tiplength=0.05
textdistance=0
barmargin=0.1
topmargin=0.1
for anno in annotations:
    if anno==annotations[-1]:
        starbars.draw_annotation([anno],fontsize=8, bar_margin=barmargin,top_margin=4*topmargin,text_distance=textdistance,tip_length=tiplength, ns_show=False)
    else:
        starbars.draw_annotation([anno],fontsize=8, bar_margin=barmargin,top_margin=topmargin,text_distance=textdistance,tip_length=tiplength, ns_show=False)
```
maybe add somthing similar to draw_annotation

**Comment by elide-b** [comment]
heyhey! Thank you for bringing this to my attention. I've looked into the problem, and it will take me some time to fix, as I will have to change some of the code structure (hopefully a week will be enough). 
I'll be back as soon as possible :)

**Comment by elide-b** [comment]
Hello hello,
thank you for your suggestion. We finally came around to fix it and in the new `2.0.0` version it won't be a problem anymore.

We upgraded the bar drawing engine to take into account the axes transformations, so even though you use logarithmic axes, it won't be a problem and the ordinary parameter values should work just the same for you! Let me know if there's any problems ✨ 

Happy plotting :)
